### ISSUE TITLE
[v2] Rule: UselessOverridingMethod

### ISSUE THREAD

**Issue by webimpress** [zendframework/zend-coding-standard#12]
Another one: 

```
FILE: zend-diactoros/src/Exception/InvalidStreamPointerPositionException.php
-----------------------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------
 17 | WARNING | Possible useless method overriding detected
    |         | (Generic.CodeAnalysis.UselessOverridingMethod.Found)
-----------------------------------------------------------------------------------------------------------------
```

and the code is like:

```php
class InvalidStreamPointerPositionException extends RuntimeException implements ExceptionInterface
{
    public function __construct(
        string $message = 'Invalid pointer position',
        $code = 0,
        ?Throwable $previous = null
    ) {
        parent::__construct($message, $code, $previous);
    }
}
```

we are changing in the constructor default value for `$message`. I think we should disable that sniff.

**Comment by webimpress** [comment]
@Ocramius Before I reported the issue in PHP_CodeSniffer I've found:
https://github.com/squizlabs/PHP_CodeSniffer/issues/519

What would you say?
Should we add manual exclusion or maybe we should change our exceptions to:
```php
class InvalidStreamPointerPositionException extends RuntimeException implements ExceptionInterface
{
    public static function invalidPointerPosition() : self
    {
        return new self('Invalid pointer position');
    }
}
```

(I think in some other places we are doing like that already).

/cc @weierophinney

**Comment by xtreamwayz** [comment]
Closed in #13

**Comment by boesing** [comment]
Is it possible that this rule found its way back to v2?

```
composer show zendframework/zend-coding-standard
name     : zendframework/zend-coding-standard
descrip. : Zend Framework Coding Standard
keywords : Coding Standard, ZendFramework, zf
versions : * 2.0.0-alpha.3
```


```
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 13 | WARNING | Possible useless method overriding detected
----------------------------------------------------------------------
```

```php
final class MyDomainException extends \DomainException
{
    private function __construct($message)
    {
        parent::__construct($message);
    }

    public static function factoryMethod() : self
    {
        return new self('Foo');
    }
}
```
